apps/studio/src/features/users/components/EditProfileModal/EditProfileModal.tsx

```diff
@@ -0,0 +1,162 @@
+import { useCallback, useEffect } from "react"
+import {
+  Button,
+  FormControl,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  VStack,
+} from "@chakra-ui/react"
+import {
+  FormErrorMessage,
+  FormLabel,
+  PhoneNumberInput,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { useAtomValue, useSetAtom } from "jotai"
+import { Controller } from "react-hook-form"
+
+import { BRIEF_TOAST_SETTINGS } from "~/constants/toast"
+import { useMe } from "~/features/me/api"
+import { useZodForm } from "~/lib/form"
+import { updateDetailsInputSchema } from "~/schemas/user"
+import { trpc } from "~/utils/trpc"
+import { updateProfileModalOpenAtom } from "../../atom"
+
+export const EditProfileModal = () => {
+  const toast = useToast(BRIEF_TOAST_SETTINGS)
+  const utils = trpc.useUtils()
+
+  const isOpen = useAtomValue(updateProfileModalOpenAtom)
+  const setIsOpen = useSetAtom(updateProfileModalOpenAtom)
+
+  const { me, isOnboarded } = useMe()
+
+  useEffect(() => {
+    if (!isOnboarded) {
+      setIsOpen(true)
+    }
+  }, [isOnboarded, setIsOpen])
+
+  const { mutate: updateDetails } = trpc.user.updateDetails.useMutation({
+    onSuccess: () => {
+      void utils.me.get.invalidate()
+      toast({
+        status: "success",
+        title: "Profile updated",
+        description: "Your profile has been updated successfully",
+      })
+      handleClose()
+    },
+    onError: (error) => {
+      toast({
+        status: "error",
+        title: "Failed to update profile",
+        description: error.message,
```

Comment:
sense check, this error will always be user-facing right (ie. no stack trace)

Reply:
yup